sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Bindings/CacheAfterWriteStream.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.Bindings
+{
+    /// <summary>
+    /// Class representing a Stream for an object that can be cached locally after being written (and before this Stream is being disposed).
+    /// This class can be used to identify if the object is eligible to be cached.
+    /// This is because the object being written is already present in shared memory.
+    /// It is used by azure-webjobs-sdk to write the object into the cache after the object is written to storage and has all the relevant
+    /// properties (e.g. Blob's eTag) populated by the azure-sdk-for-net storage extension.
+    /// </summary>
+    internal class CacheAfterWriteStream : Stream, ICacheAfterWriteStream
```

Comment:
The `Stream` and `ICacheAfterWriteStream` seem to be quite orthogonal concepts. Combining them like this make this type not very cohesive. One sign of this is that there's nothing in this class that would mix `Stream` and `ICacheAfterWriteStream`.

If the goal is to carry extra data along the stream then I'd suggest to use composition, i.e. create type enclosing Stream and these extra properties. This might involve some work on the binding configuration but would be less hacky.